src/subscribe.rs

```diff
@@ -0,0 +1,429 @@
+#[ft_sdk::form]
+fn subscribe(
+    ft_sdk::Query(name): ft_sdk::Query<"name", Option<String>>,
+    ft_sdk::Query(phone): ft_sdk::Query<"phone", Option<String>>,
+    ft_sdk::Query(email): ft_sdk::Query<"email", Option<String>>,
+    ft_sdk::Query(source): ft_sdk::Query<"source", Option<String>>,
+    ft_sdk::Query(topic): ft_sdk::Query<"topic", Option<String>>,
+    ft_sdk::Query(next): ft_sdk::Query<"next", Option<String>>,
+    sid: ft_sdk::Cookie<{ ft_sdk::auth::SESSION_KEY }>,
+    host: ft_sdk::Host,
+    mountpoint: ft_sdk::Mountpoint,
+    mut conn: ft_sdk::Connection,
+) -> ft_sdk::form::Result {
+    use diesel::prelude::*;
+
+    let (subscriber, id) = validate(&mut conn, name, phone, email, sid)?;
+    let user_id = get_or_create_user_id(&mut conn, id, &subscriber)?;
+
+    #[derive(diesel::QueryableByName)]
+    #[diesel(table_name = ft_sdk::auth::fastn_user)]
+    struct UserData {
+        data: String,
+    }
+
+    ft_sdk::println!("querying data");
+    let data: serde_json::Value = match diesel::sql_query(
+        r#"
+        SELECT data from fastn_user where id = $1 LIMIT 1;
+        "#,
+    )
+    .bind::<diesel::sql_types::Integer, _>(user_id.0)
+    .get_result::<UserData>(&mut conn)
+    {
+        Ok(d) => serde_json::from_str(&d.data)?,
+        Err(e) => return Err(e.into()),
+    };
```

Comment:
We need subscription data only. Use `ft_sdk::auth::provider::user_data_by_id` to fetch and `ft_sdk::auth::provider::update_user` to update.

Reply:
> We need subscription data only
True. But most of the subscription data is incompatible with `ft_sdk::auth::ProviderData` returned by these `user_data_*` functions.

There are only a few keys that the subscription app interacts with (`subscribed`, `topics`, `sources`, `confirmed` etc) but there can be arbitrary keys here that may have been imported from other sources (benefits of schemaless). Mapping these keys to a struct that is nothing but just a bunch of `Option`s and then using `update` ensuring that the old values are preserved does not sound simpler or cleaner than the approach I've used here. I strongly feel that doing it your way would make this code any better. We can come back to this and improve it definitely in the future but I want to focus on getting the things that work to prod first.